### ISSUE TITLE
CMake support?

### ISSUE THREAD

**Issue by pierre-aimi** [ckormanyos/xxd#16]
Would you consider a CMakeLists.txt file in your repo to offer CMake compilation support?

I just used this one to compile on Windows, and it should work without modification on Linux/MacOS I think.

```
cmake_minimum_required(VERSION 3.1 FATAL_ERROR)

set(CMAKE_C_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED on)

project(xxd LANGUAGES C CXX)

file(GLOB SOURCES
    "src/*.c"
)

set(EXE xxd)

add_executable(${EXE} ${SOURCES}
```

On Windows you need to set up a MSVC environment in a  command prompt - e.g. for VS2022 I ran "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Auxiliary\Build\vcvars64.bat"

then it's just 
```
mkdir build
cd build
cmake .. -G Ninja && ninja
```
Having CMake support would allow us to better integrate xxd

**Comment by ckormanyos** [comment]
Sure I’d be very happy to support a CMake build. I’ll be absent for the next three weeks or so, which means this will probably wait until then. But when I return, let’s try to get a CMake build running and commuted and I’d also like it in a few Cl jobs. Thanks for this suggestion

**Comment by pierre-aimi** [comment]
Ok great! Ping me when you are back

**Comment by ckormanyos** [comment]
Hi @pierre-aimi I have returned to development and just approved CI in your PR, #17

**Comment by ckormanyos** [comment]
Hello @pierre-aimi this is now fixed by #17 and merged to main. Thank you for adding this feature.
